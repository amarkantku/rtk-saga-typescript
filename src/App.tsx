import './App.css';
import Checky from './components/Checky/Checky';
import Select from './components/Select/Select';

type ReactComponent = JSX.Element;

const App: React.FC = (): ReactComponent => (
  <div className='App'>
    <header className='App-header' role={'banner'}>
      <h1>Learning React Testing Library</h1>
    </header>
    <section>
      <div
        style={{
          display: 'flex',
          flexDirection: 'column',
          alignItems: 'center',
          justifyContent: 'flex-start',
          width: '80%',
          height: '75px',
          margin: '0 auto',
          border: '1px solid #eb1010',
          background: '#cdcdcd',
          padding: '20px 0',
        }}
      >
        <Select />
      </div>
      <div
        style={{
          display: 'flex',
          flexDirection: 'column',
          alignItems: 'center',
          justifyContent: 'flex-start',
          width: '80%',
          height: '75px',
          margin: '0 auto',
          border: '1px solid #3e32ac',
          background: '#ebd2d2',
          padding: '20px 0',
        }}
      >
        <Checky />
      </div>
    </section>
  </div>
);

export default App;
