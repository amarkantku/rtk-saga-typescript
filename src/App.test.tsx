import { render, cleanup } from '@testing-library/react';
import { Provider } from 'react-redux';
import { store } from './app/store';
import App from './App';

describe('<App /> test suite', () => {
  afterEach(cleanup)

  test('should render correctly', () => {
    const { container } = render(
      <Provider store={store}>
        <App />
      </Provider>
    );
    expect(container).toMatchSnapshot();
  });

  test('should render learning react testing library', () => {
    const { getByText, getByRole, container } = render(
      <Provider store={store}>
        <App />
      </Provider>
    );
    expect(getByText(/learning react testing library/i)).toBeInTheDocument();
    expect(getByRole('banner')).toBeInTheDocument();
  });
});
