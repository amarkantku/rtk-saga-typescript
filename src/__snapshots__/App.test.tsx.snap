// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> test suite should render correctly 1`] = `
<div>
  <div
    class="App"
  >
    <header
      class="App-header"
      role="banner"
    >
      <h1>
        Learning React Testing Library
      </h1>
    </header>
    <section>
      <div
        style="display: flex; flex-direction: column; align-items: center; justify-content: flex-start; width: 80%; height: 75px; margin: 0px auto; border: 1px solid #eb1010; background: rgb(205, 205, 205); padding: 20px 0px;"
      >
        <div>
          <select
            role="combobox"
          >
            <option
              value=""
            >
              Select a country
            </option>
            <option
              value="AT"
            >
              Austria
            </option>
            <option
              value="IE"
            >
              Ireland
            </option>
            <option
              value="US"
            >
              United States
            </option>
          </select>
          <p />
        </div>
      </div>
      <div
        style="display: flex; flex-direction: column; align-items: center; justify-content: flex-start; width: 80%; height: 75px; margin: 0px auto; border: 1px solid #3e32ac; background: rgb(235, 210, 210); padding: 20px 0px;"
      >
        <div>
          <h2>
            Checky
          </h2>
          <input
            id="checky2"
            type="checkbox"
          />
          <label
            for="checky2"
          >
            false
          </label>
        </div>
      </div>
    </section>
  </div>
</div>
`;
